kasa/discover.py

```diff
@@ -580,19 +603,26 @@ async def try_connect_all(
             and (protocol := get_protocol(config))
             and (
                 device_class := get_device_class_from_family(
-                    device_family.value, https=https
+                    device_family.value, https=https, no_warn=True
                 )
             )
         }
-        for protocol, config in candidates.values():
+        for key, val in candidates.items():
             try:
-                dev = await _connect(config, protocol)
+                prot, config = val
+                dev = await _connect(config, prot)
             except Exception:
-                _LOGGER.debug("Unable to connect with %s", protocol)
+                _LOGGER.debug("Unable to connect with %s", prot)
+                if on_attempt:
+                    ca = tuple.__new__(ConnectAttempt, key)
+                    on_attempt(ca, False)
```

Comment:
Is it possible to pass the parameters as kwarg-onlys? That'd improve the readability.

Reply:
On line 617 or 618?